File: ios/VoiceNotesOndevice/Models/ExportFormat.swift
import Foundation

/// Available export formats for transcripts
enum ExportFormat: String, CaseIterable, Identifiable {
    case txt = "txt"
    case srt = "srt"
    case vtt = "vtt"

    var id: String { rawValue }

    /// Human-readable display name
    var displayName: String {
        switch self {
        case .txt: return "Plain Text (.txt)"
        case .srt: return "SubRip Subtitle (.srt)"
        case .vtt: return "WebVTT (.vtt)"
        }
    }

    /// Short name for UI
    var shortName: String {
        rawValue.uppercased()
    }

    /// File extension
    var fileExtension: String { rawValue }

    /// MIME type for sharing
    var mimeType: String {
        switch self {
        case .txt: return "text/plain"
        case .srt: return "application/x-subrip"
        case .vtt: return "text/vtt"
        }
    }

    /// Description of the format
    var formatDescription: String {
        switch self {
        case .txt:
            return "Simple text without timestamps. Best for reading and editing."
        case .srt:
            return "Industry-standard subtitle format. Compatible with most video editors."
        case .vtt:
            return "Web-compatible subtitle format. Works with HTML5 video players."
        }
    }
}

File: windows/src/SecureVox.Core/Models/ExportFormat.cs
namespace SecureVox.Core.Models;

/// <summary>
/// Supported export formats for transcripts
/// </summary>
public enum ExportFormat
{
    Txt,
    Srt,
    Vtt,
    Json
}

public static class ExportFormatExtensions
{
    public static string GetDisplayName(this ExportFormat format) => format switch
    {
        ExportFormat.Txt => "Plain Text (.txt)",
        ExportFormat.Srt => "SubRip Subtitle (.srt)",
        ExportFormat.Vtt => "WebVTT (.vtt)",
        ExportFormat.Json => "JSON (.json)",
        _ => format.ToString()
    };

    public static string GetFileExtension(this ExportFormat format) => format switch
    {
        ExportFormat.Txt => ".txt",
        ExportFormat.Srt => ".srt",
        ExportFormat.Vtt => ".vtt",
        ExportFormat.Json => ".json",
        _ => ".txt"
    };

    public static string GetMimeType(this ExportFormat format) => format switch
    {
        ExportFormat.Txt => "text/plain",
        ExportFormat.Srt => "application/x-subrip",
        ExportFormat.Vtt => "text/vtt",
        ExportFormat.Json => "application/json",
        _ => "text/plain"
    };
}
